### PR TITLE
remove PLUGIN_VERSION_STRING macro from prefix files

### PR DESCRIPTION
**Pull request by sophiapoirier**
There was a historical reason for the separate macro definitions for integers `PLUGIN_VERSION_MAJOR`, `PLUGIN_VERSION_MINOR`, and `PLUGIN_VERSION_BUGFIX` vs the string `PLUGIN_VERSION_STRING` and now can't remember which resource compiler(s) was problematic. Once upon a time, there was Microsoft's RC and Apple's Rez, but now only RC is in the picture. I am curious if it is the better behaved one and if its preprocessor can handle doing the macro stringification and concatenation? If so then would be nice to remove some redundancy in `#define`s.

### CODE REVIEW

File: dfx-library/visualstudio/dfxplugin-version.rc [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #endif
 
 #define PLUGIN_DISPLAY_NAME	PLUGIN_NAME_STRING
+#define PLUGIN_VERSION_STRING #PLUGIN_VERSION_MAJOR #PLUGIN_VERSION_MINOR #PLUGIN_VERSION_BUGFIX
```
Comment (tom7) [inline dfx-library/visualstudio/dfxplugin-version.rc]: Indeed this doesn't compile (turns out mingw's windres just uses gcc to preprocess). Two problems: I think it is only actually legal to #stringify macro arguments, not already-existing defines. But if you try like #define CONCAT(a, b) #a "." #b and calling CONCAT(MAJOR, MINOR) then you just get a string like "MAJOR.MINOR". This page explains, but also the solution of using two macros:
https://gcc.gnu.org/onlinedocs/gcc-3.4.6/cpp/Argument-Prescan.html#Argument-Prescan

**Review by tom7 — APPROVED** [review]
Thanks! Looks good with my fixes; everything compiles for me and I suspect this has no effect on the mac builds?

File: dfx-library/visualstudio/dfxplugin-version.rc (hunk 1 above)
Comment (sophiapoirier) [reply dfx-library/visualstudio/dfxplugin-version.rc]: Ah yes, I remember having to do this once upon a time for generating unique AU View class names in:
https://github.com/sophiapoirier/destroyfx/blob/8302de1fbd98a52125995d7bf6b4b835ecdacae7/dfxgui/dfxgui-auviewfactory.mm#L35
Thanks for the multiple fixes!

**Comment by sophiapoirier** [comment]
Thank you and yes, no effect on Mac builds (once upon a time it related, but no longer).

**Pull request merged**